spec/infinispan_cluster_spec.js

```diff
@@ -30,11 +33,467 @@ describe('Infinispan cluster client', function() {
       .catch(t.failed(done)).finally(done);
   });
 
+  it('can put a key/value pair using different clients', function(done) {
```

Comment:
Actually, regardless of whether it's distributed or replicated, if the client is hash topology aware it'll get hash topology for both so it'll route it according to that.

It's interesting to understand that at the client level, what's interesting is how it does the routing. That's why executing iterator against a cluster of servers is interesting because the way it should route is that it should be sticky. So, whichever node where the iterator gets started, it should direct the rest of reqs for that node to the same node. However, the current code does round robin so that's why iterator w/ cluster fails ([JDG-184](https://issues.jboss.org/browse/JDG-184)).

By explaining this what I'm trying to get to is that it's not wise to add tests for the shake of it unless their really testing something that will highlight client issues. That's why I'm not 100% sure having multi-client tests will add a lot to the testsuite. I know there are some tests that fail in your PR, but I need to inspect them closely to see if they can be replicated with a single client because each client created should be independent (each has their own independent components).

IOW, if you can replicate the same issue with 1 client, it's easier to debug/reproduce so we should strive to do that. If there really are tests that fail with multiple clients, we should look and understand what's the reason behind it and then decide to add more such tests.

Having said that, I'm happy to integrate the tests you have so far with multiple clients, but might refactor them later once I've spotted the root cause.